### PR TITLE
Late deserialization and some bug fixing

### PR DESCRIPTION
**Pull request by VishnAndr**
Late deserialization for TOBJ ( https://github.com/larshp/abapGit/issues/206 ) , FUGR had a bug regarding dynpro, ENHO should be deleted if exists before deserializiation,.

### CODE REVIEW

File: zabapgit.prog.abap [hunk 1]
```diff
@@ -1205,6 +1207,7 @@ CLASS lcl_xml IMPLEMENTATION.
 *    buffer serializer metadata. Git node will be removed lateron
     ms_metadata-class = li_element->get_attribute_ns( c_attr_serializer ).
     ms_metadata-version = li_element->get_attribute_ns( c_attr_serializer_version ).
+    ms_metadata-late_deser = li_element->get_attribute_ns( c_attr_late_deserialize ).
```
Comment (larshp) [inline zabapgit.prog.abap]: I dont think we need the flag in the serialized XML, we can just let the code always decide if it should do late deserialization.

File: zabapgit.prog.abap [hunk 2]
```diff
@@ -1369,6 +1372,8 @@ CLASS lcl_xml_output IMPLEMENTATION.
                              value = is_metadata-class ).   "#EC NOTEXT
       li_git->set_attribute( name  = c_attr_serializer_version
                              value = is_metadata-version ). "#EC NOTEXT
+      li_git->set_attribute( name  = c_attr_late_deserialize
+                             value = is_metadata-late_deser ). "#EC NOTEXT
```
Comment (larshp) [inline zabapgit.prog.abap]: see above comment

File: zabapgit.prog.abap [hunk 3]
```diff
@@ -13307,6 +13322,42 @@ CLASS lcl_objects IMPLEMENTATION.
 
     lcl_objects_activation=>activate( ).
 
+
+    LOOP AT lt_late_deser_results ASSIGNING <ls_result>.
```
Comment (larshp) [inline zabapgit.prog.abap]: can we move this loop to a new method, which imports lt_late_deser_results? also it looks similar to the LOOP AT lt_results above, so perhaps that loop can also be replaced with the new method call

**Comment by larshp** [comment]
cool, thanks, have a few comments, see above

File: zabapgit.prog.abap (hunk 1 above)
Comment (larshp) [inline zabapgit.prog.abap]: I dont think we need the flag in the serialized XML, we can just let the code always decide if it should do late deserialization.

Call the create_object method, on this method the get_metadata can be called to get the late deserialization flag, without saving it in the XML

File: zabapgit.prog.abap (hunk 2 above)
Comment (larshp) [inline zabapgit.prog.abap]: see above comment

**Comment by larshp** [comment]
will merge and do the changes

**Comment by VishnAndr** [comment]
Great! Thanks a lot!!! /had no time during the week - appreciate your help!/

**Pull request merged**